src/main/java/sora/task/TaskList.java

```diff
@@ -0,0 +1,286 @@
+        int taskNumber;
+        try {
+            taskNumber = Integer.parseInt(command.substring(7)) - 1;
+        } catch (NumberFormatException e) {
+            throw new TaskNotFoundException();
+        }
+
+        // Throw exception if taskNumber is out of range
+        if (taskNumber < 0 || taskNumber >= tasks.size()) {
+            throw new TaskNotFoundException();
+        }
+
+        // Delete the task
+        Task task = tasks.remove(taskNumber);
+
+        int size = tasks.size();
+
+        // Return confirmation message
+        return Message.DELETE + "\n  "
+            + task + "\n" + (
+            size == 0
+            ? "You have no more task in the list!"
+            : size == 1
+              ? "You still have 1 task in the list."
+              : "You still have " + tasks.size() + " tasks in the list.");
+    }
+
+    /**
+     * Marks the task with the same task number specified by user as done.
+     *
+     * @param command command entered by user (done [task number])
+     * @return confirmation message for successfully marked a task as done
+     * @throws IllegalFormatException if user gives empty or invalid task number
+     * @throws TaskNotFoundException if the task specified by the task number does not exist
+     */
+    public String taskDone(String command) throws IllegalFormatException, TaskNotFoundException {
+        // Throw exception if user gives empty or non-integer task number
+        validateCommand(command, "^done [0-9]+", "done [task number]");
+
+        int taskNumber;
+        try {
+            taskNumber = Integer.parseInt(command.substring(5)) - 1;
+        } catch (NumberFormatException e) {
+            throw new TaskNotFoundException();
+        }
+
+        // Throw exception if taskNumber is out of range
+        if (taskNumber < 0 || taskNumber >= tasks.size()) {
+            throw new TaskNotFoundException();
+        }
+
+        // Mark the task as done
+        Task task = tasks.get(taskNumber);
+        task.markAsDone();
+
+        // Return confirmation message
+        return Message.DONE + "\n  " + task;
+    }
+
+    /**
+     * Adds a new event to the list of tasks.
+     *
+     * @param command command entered by user (event [description] /at [dd/MM/yy] /from [HHmm] /to [HHmm])
+     * @return confirmation message for successfully added a new event
+     * @throws IllegalFormatException if user inputs an invalid command
+     */
+    public String addEvent(String command) throws IllegalFormatException {
+        String correctFormat = "event [description] /at [dd/MM/yy] /from [HHmm] /to [HHmm]";
+
+        // Throw exception if command does not follow format
+        validateCommand(command, "^event .* /at .* /from .* /to .*", correctFormat);
+
+        // Separate info and trim each of them
+        String[] info = command.substring(6).split("/at|/from|/to");
+        info = Arrays.stream(info)
+            .map(String::trim)
+            .toArray(String[]::new);
```

Comment:
I'm not very sure about this, but I think need to add 4 space indentation to this line?

Reply:
I'm not really sure about the indentation rule here. Setting it to 8 violates checkstyle rules so I had to switch everything down to 4 which somehow checkstyle is not giving me any troubles. Find it really hard to configure the specific rules so I'm being kinda lazy there.

Reply:
Anyways i went to fix it, add some stuffs to suppress warnings haha